src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -545,6 +551,69 @@ private ImmutableArray<AdditionalTextFile> ResolveAdditionalFilesFromArguments(L
+        {
+            Debug.Assert(analyzerDriver.AnalyzerExecutionTimes != null);
+            if (analyzerDriver.AnalyzerExecutionTimes.IsEmpty)
+            {
+                return;
+            }
+
+            var totalAnalyzerExecutionTime = analyzerDriver.AnalyzerExecutionTimes.Sum(kvp => kvp.Value.TotalSeconds);
+            Func<double, string> getFormattedTime = d => d.ToString("##0.000");
+            consoleOutput.WriteLine();
+            consoleOutput.WriteLine(string.Format(CodeAnalysisResources.AnalyzerTotalExecutionTime, getFormattedTime(totalAnalyzerExecutionTime)));
+
+            if (isConcurrentBuild)
+            {
+                consoleOutput.WriteLine(CodeAnalysisResources.MultithreadedAnalyzerExecutionNote);
+            }
+
+            var analyzersByAssembly = analyzerDriver.AnalyzerExecutionTimes
+                .GroupBy(kvp => kvp.Key.GetType().GetTypeInfo().Assembly)
+                .OrderByDescending(kvp => kvp.Sum(entry => entry.Value.Ticks));
+
+            consoleOutput.WriteLine();
+
+            getFormattedTime = d => d < 0.001 ? string.Format("{0,8}", "<0.001") : string.Format("{0,8:##0.000}", d);
+            Func<int, string> getFormattedPercentage = i => string.Format("{0,5}", (i < 1 ? "<1" : i.ToString()));
+            Func<string, string> getFormattedAnalyzerName = s => "   " + s;
+
+            // Table header
+            var analyzerTimeColumn = string.Format("{0,8}", CodeAnalysisResources.AnalyzerExecutionTimeColumnHeader);
+            var analyzerPercentageColumn = string.Format("{0,5}", "%");
+            var analyzerNameColumn = getFormattedAnalyzerName(CodeAnalysisResources.AnalyzerNameColumnHeader);
+            consoleOutput.WriteLine(string.Format(CodeAnalysisResources.AnalyzerExecutionTimeRow, analyzerTimeColumn, analyzerPercentageColumn, analyzerNameColumn));
+
+            // Table rows grouped by assembly.
+            foreach (var analyzerGroup in analyzersByAssembly)
+            {
+                var executionTime = analyzerGroup.Sum(kvp => kvp.Value.TotalSeconds);
+                var percentage = (int)(executionTime * 100 / totalAnalyzerExecutionTime);
+
+                analyzerTimeColumn = getFormattedTime(executionTime);
+                analyzerPercentageColumn = getFormattedPercentage(percentage);
+                analyzerNameColumn = getFormattedAnalyzerName(analyzerGroup.Key.FullName);
+
+                consoleOutput.WriteLine(string.Format(CodeAnalysisResources.AnalyzerExecutionTimeRow, analyzerTimeColumn, analyzerPercentageColumn, analyzerNameColumn));
+
+                // Rows for each diagnostic analyzer in the assembly.
+                foreach (var kvp in analyzerGroup.OrderByDescending(kvp => kvp.Value))
+                {
+                    executionTime = kvp.Value.TotalSeconds;
+                    percentage = (int)(executionTime * 100 / totalAnalyzerExecutionTime);
```

Comment:
No I meant, if no analyzers were there for matching project language or say all analyzers can't be instantiated or throw up front. Some weird case probably, but I have a check upfront in the method, just in case.